skipper.go

```diff
@@ -1200,6 +1258,33 @@ func run(o Options, sig chan os.Signal, idleConnsCH chan struct{}) error {
 		)
 	}
 
+	oauthConfig := &auth.OAuthConfig{}
+	if o.EnableOAuth2GrantFlow /* explicitly enable grant flow */ {
+		oauthConfig.AuthURL = o.OAuth2AuthURL
+		oauthConfig.TokenURL = o.OAuth2TokenURL
+		oauthConfig.TokeninfoURL = o.OAuthTokeninfoURL
+		oauthConfig.SecretFile = o.OAuth2SecretFile
+		oauthConfig.ClientID = o.OAuth2ClientID
+		oauthConfig.ClientSecret = o.OAuth2ClientSecret
+		oauthConfig.ClientIDFile = o.OAuth2ClientIDFile
+		oauthConfig.ClientSecretFile = o.OAuth2ClientSecretFile
+		oauthConfig.CallbackPath = o.OAuth2CallbackPath
+		oauthConfig.AuthURLParameters = o.OAuth2AuthURLParameters
+		oauthConfig.SecretsProvider = sp
+		oauthConfig.Secrets = secrets.NewRegistry()
```

Comment:
we can also use `o.SecretsRegistry` or do I miss something?

Reply:
Looks legit.

Unrelated: just wondering the Registry.GetEncrypter() is not synchronized, shouldn't it be? See: https://github.com/zalando/skipper/blob/master/secrets/registry.go#L29

Reply:
Yes, thanks will change it on Monday

Reply:
was merged